lib/schema/validate.js

```diff
@@ -122,22 +122,41 @@ const processFieldNames = ({ payload = {}, nesting = [], schema = {}, results =
     const value = payload[f];
 
     if (Array.isArray(value)) {
-      value.forEach(
-        v => v instanceof Object && processFieldNames({ payload: v, nesting: [f, 'properties'], schema, results, resourceName })
-      );
+      const updatedNestedPayloads = [];
+      value.forEach(v => {
+        if (v instanceof Object) {
+          const { payload: updatedNestedPayload } = processFieldNames({
+            payload: v,
+            nesting: [f, 'properties'],
+            schema,
+            results,
+            resourceName
+          });
+          updatedNestedPayloads.push(updatedNestedPayload);
+        } else {
+          updatedNestedPayloads.push(v);
+        }
+      });
+      payload = {
```

Comment:
This is still mutating `payload`, which is passed in as parameter to a recursive function. This should be changed (and the workaround shouldn't be `structuredClone`).

We should also not be using recursion in the codebase unless there's some specific reason. I can't see one here.